hs_core/tests/api/views/test_group_share.py

```diff
@@ -0,0 +1,451 @@
+        # let john reject lisa's request
+        self._check_act_on_membership_request(self.john, membership_request, 'reject')
+        # there should not be any pending requests at this point
+        self.assertEqual(len(self.test_group.gaccess.group_membership_requests), 0)
+        # test lisa is not a member of test_group
+        self.assertNotIn(self.lisa, self.test_group.gaccess.members)
+
+    def test_act_on_membership_request_failure(self):
+        # here we are testing act_on_group_membership_request view function
+        # group owner accepting/rejecting user request to join a group
+
+        # test accepting user request to join failure
+
+        # there should not be any pending requests at this point
+        self.assertEqual(len(self.test_group.gaccess.group_membership_requests), 0)
+        # let mike request to join test group
+        self.mike.uaccess.create_group_membership_request(self.test_group)
+        # there should 1 pending request at this point
+        self.assertEqual(len(self.test_group.gaccess.group_membership_requests), 1)
+        membership_request = self.test_group.gaccess.group_membership_requests.first()
+        # let lisa (not a group owner) try to accept mike's request - should fail
+        self._check_act_on_membership_request(self.lisa, membership_request, 'accept',
+                                              success=False)
+        # there should be 1 pending requests at this point
+        self.assertEqual(len(self.test_group.gaccess.group_membership_requests), 1)
+        # test mike is not a member of test_group
+        self.assertNotIn(self.mike, self.test_group.gaccess.members)
+
+    def _check_share_with_user(self, privilege):
+
+        url_params = {'group_id': self.test_group.id, 'privilege': privilege,
+                      'user_id': self.mike.id}
+        url = reverse('share_group_with_user', kwargs=url_params)
+        request = self.factory.post(url, data={})
+        request.user = self.john
+        request.META['HTTP_REFERER'] = '/group/{}'.format(self.test_group.id)
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = share_group_with_user(request, group_id=self.test_group.id,
+                                         privilege=privilege, user_id=self.mike.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        success_messages = [m for m in flag_messages if m.tags == 'success']
+        self.assertNotEqual(len(success_messages), 0)
+        self.test_group.gaccess.refresh_from_db()
+
+    def _check_membership_request(self, invitation=True):
+
+        user_id = None
+        if invitation:
+            user_id = self.mike.id
+            user = self.john
+        else:
+            user = self.mike
+
+        if user_id is not None:
```

Comment:
There is a difference what I have and what you are suggesting. The conditional expression you proposing works for ```None``` as well as ```True/False```.  The expression that I have only cares for ```None``` and more explicit condition.